CrossAppLogin/Back/src/main/kotlin/com/infomaniak/core/crossapplogin/back/BaseCrossAppLoginViewModel.kt

```diff
@@ -40,18 +50,64 @@ import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.transform
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.plus
 import kotlinx.serialization.ExperimentalSerializationApi
+import okhttp3.OkHttpClient
 import com.infomaniak.core.R as RCore
 import com.infomaniak.core.appintegrity.R as RAppIntegrity
 import com.infomaniak.core.network.R as RCoreNetwork
 
+private val apiRepository = object : ApiRepositoryCore() {}
+
+private val baseOkHttpClient by lazy {
+    OkHttpClient.Builder().apply {
+        addCache()
+        addCommonInterceptors()
+    }
+}
```

Comment:
Any reason you didn't put those into the class?

I don't think they're particularly heavy, but still, they can't be reclaimed by the GC after they're no longer needed, and I don't think we need to optimize coming back multiple times to CrossAppLogin, I'd say it's highly likely to be a one-off operation in the process lifetime (besides QA testing).